rest/test/routes/transactionRoutes_spec.js

```diff
@@ -18,96 +18,275 @@
-			addGetPostTests(
-				'transactionsByHashes',
-				'transactionId',
-				Valid_Transaction_Hashes,
-				Valid_Transaction_Hashes.map(convert.hexToUint8)
-			);
-			addHomogeneousCheck(Valid_Transaction_Hashes, Valid_Object_Ids[0]);
+				testCases.forEach(testCase => {
+					runParseFilterTest(testCase.filter, testCase.param, testCase.value);
+				});
+
+				it('transactionTypes', () => {
+					const req = { params: { type: ['1', '5', '25'] } };
+
+					// Act + Assert
+					return mockServer.callRoute(route, req).then(() => {
+						expect(dbTransactionsFake.firstCall.args[0]).to.deep.equal({
+							address: undefined,
+							height: undefined,
+							recipientAddress: undefined,
+							signerPublicKey: undefined,
+							state: undefined,
+							transactionTypes: [1, 5, 25]
+						});
+					});
+				});
+			});
+
+			describe('parses options', () => {
+				it('parses options correctly', () => {
+					const req = {
+						params: {
+							pageSize: '15', pageNumber: '5', sortField: 'meta.hash', order: 'desc'
+						}
+					};
+
+					// Act + Assert
+					return mockServer.callRoute(route, req).then(() => {
+						expect(dbTransactionsFake.firstCall.args[1]).to.deep.equal({
+							pageSize: 15,
+							pageNumber: 5,
+							sortField: 'meta.hash',
+							sortDirection: -1
+						});
+					});
+				});
+			});
+
+			describe('does not allow filtering by address if signerPublicKey or recipientAddress are provided', () => {
+				const errorMessage = 'can\'t filter by address if signerPublicKey or recipientAddress are already provided';
+
+				it('address and signer public key', () => {
+					const req = {
+						params: { address: testAddressString, signerPublicKey: testPublickeyString }
+					};
+
+					// Act + Assert
+					expect(() => mockServer.callRoute(route, req)).to.throw(errorMessage);
+				});
+
+				it('address and recipient address', () => {
+					const req = {
+						params: { address: testAddressString, recipientAddress: testAddressString }
+					};
+
+					// Act + Assert
+					expect(() => mockServer.callRoute(route, req)).to.throw(errorMessage);
+				});
+			});
+
+			describe('checks correct state is provided', () => {
+				const runValidStateTest = state => {
+					it(state, () =>
+						// Act + Assert
+						mockServer.callRoute(route, { params: { state } }).then(() => {
+							expect(dbTransactionsFake.firstCall.args[0]).to.deep.equal({
+								address: undefined,
+								height: undefined,
+								recipientAddress: undefined,
+								signerPublicKey: undefined,
+								state,
+								transactionTypes: undefined
+							});
+						}));
+				};
+
+				['confirmed', 'unconfirmed', 'partial'].forEach(state => runValidStateTest(state));
+
+				it('invalid', () => {
+					const req = {
+						params: { state: 'nonsenseSatate' }
+					};
+
+					// Act + Assert
+					expect(() => mockServer.callRoute(route, req)).to.throw('invalid transaction state provided');
+				});
+			});
 		});
 	});
 });
```

Comment:
forced `sortField` to `id` now, this must be adapted in the tests, and also have a specific test that shows this behaviour (always defaults sortField to id)